### PR TITLE
test: Replace deprecated method with non-deprecated one

### LINKED ISSUE #278: test: Replace deprecated method with non-deprecated one
**Is your feature request related to a problem? Please describe.**
Since we have several deprecated method in the ITPubSubTest.

**Describe the solution you'd like**
I think we can begin to migrate it to non-deprecated method one.

**Describe alternatives you've considered**
Any alternative solutions or features you've considered.

**Additional context**
Any other context or screenshots about the feature request.

### PR DESCRIPTION
**Pull request by irvifa**
Since we have several deprecated method in the ITPubSubTest,
I think we can begin to migrate it to non-deprecated method one.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-pubsub/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #278 ☕️

### CODE REVIEW

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java [hunk 1]
```diff
@@ -115,19 +142,23 @@ public void testTopicPolicy() {
   @Test
   public void testVPCPushSubscriber() {
     assumeTrue(IS_VPC_TEST);
-    ProjectTopicName topicName =
-        ProjectTopicName.of(projectId, formatForTest("testing-vpc-push-subscriber-topic"));
+    TopicName topicName =
+        TopicName.newBuilder()
+            .setProject(projectId)
+            .setTopic(formatForTest("testing-topic-policy"))
```
Comment (hannahrogers-google) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java:148]: Let's keep the same topic name here: "testing-vpc-push-subscriber-topic"

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java [hunk 2]
```diff
@@ -89,23 +93,46 @@ private String formatForTest(String resourceName) {
     return resourceName + "-" + NAME_SUFFIX;
   }
 
+  private Subscription getSubscription(
+      String subscriptionName, String topicName, PushConfig pushConfig, int ackDeadline) {
```
Comment (hannahrogers-google) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java:97]: Can we take in SubscriptionName and TopicName here, and do the toString conversions within this method?

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java [hunk 3]
```diff
@@ -139,15 +170,23 @@ public void testVPCPushSubscriber() {
 
   @Test
   public void testPublishSubscribe() throws Exception {
-    ProjectTopicName topicName =
-        ProjectTopicName.of(projectId, formatForTest("testing-publish-subscribe-topic"));
+    TopicName topicName =
+        TopicName.newBuilder()
+            .setProject(projectId)
+            .setTopic(formatForTest("testing-topic-policy"))
```
Comment (hannahrogers-google) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java:176]: Let's keep the same topic name here as well: "testing-publish-subscribe-topic"

**Comment by irvifa** [comment]
HI @hannahrogers-google I already amend review suggestions on the latest commit.

**Comment by irvifa** [comment]
@hannahrogers-google I fixed the method by adding the bindings:
```
policy.toBuilder().addBindings(binding).build()
```

**Comment by hannahrogers-google** [comment]
Thanks @irvifa, the PR looks good to me. A couple of tests seem to be stuck in the 'Expected — Waiting for status to be reported' state. Can you try pushing a fresh commit to hopefully resolve this issue?

**Comment by irvifa** [comment]
@hannahrogers-google yep, I tried to push it again, hopefully it won't got stuck this time.

**Comment by hannahrogers-google** [comment]
> @hannahrogers-google yep, I tried to push it again, hopefully it won't got stuck this time. However it will need another `kokoro` test again.

thanks, looks like they are all running this time :) I will merge when the tests complete

**Pull request merged**